src/disk.c

```diff
@@ -1059,26 +1080,18 @@ static int disk_read(void) {
     }
 
     metric_t m = {0};
-    metric_label_set(&m, "device", ksp[i]->ks_name);
+    metric_label_set(&m, device_label, ksp[i]->ks_name);
 
-    metric_family_append(&fam_io, "direction", "read",
+    metric_family_append(&fam_io, direction_label, read_direction,
                          (value_t){.counter = kio.KIO_ROCTETS}, &m);
-    metric_family_append(&fam_io, "direction", "write",
+    metric_family_append(&fam_io, direction_label, write_direction,
                          (value_t){.counter = kio.KIO_WOCTETS}, &m);
 
-    metric_family_append(&fam_ops, "direction", "read",
+    metric_family_append(&fam_ops, direction_label, read_direction,
                          (value_t){.counter = kio.KIO_ROPS}, &m);
-    metric_family_append(&fam_ops, "direction", "write",
+    metric_family_append(&fam_ops, direction_label, write_direction,
                          (value_t){.counter = kio.KIO_WOPS}, &m);
 
-    if (strncmp(ksp[i]->ks_class, "disk", 4) == 0) {
-      /* FIXME: Convert this to microseconds if necessary */
-      metric_family_append(&fam_time, "direction", "read",
-                           (value_t){.counter = kio.KIO_RTIME}, &m);
-      metric_family_append(&fam_time, "direction", "write",
-                           (value_t){.counter = kio.KIO_WTIME}, &m);
-    }
-
```

Comment:
You commented in commit removing these Solaris metrics, them being run & wait, instead of read & write => maybe they should now use `fam_disk_io_time`?

Note: wait time is important disk metric.  It tells how much things are stalling because there's too much to write / disk is too slow.

Reply:
I have added a few metrics for Solaris based on the kstat docs.